tests/components/deconz/test_services.py

```diff
@@ -198,3 +213,75 @@ async def test_service_refresh_devices(hass):
         "scene.group_1_name_scene_1": "/groups/1/scenes/1",
         "sensor.sensor_1_name": "/sensors/1",
     }
+
+
+async def test_remove_orphaned_entries_service(hass):
+    """Test service works and also don't remove more than expected."""
+    data = deepcopy(DECONZ_WEB_REQUEST)
+    data["lights"] = deepcopy(LIGHT)
+    data["sensors"] = deepcopy(SWITCH)
+    gateway = await setup_deconz_integration(hass, get_state_response=data)
+
+    data = {CONF_BRIDGE_ID: BRIDGEID}
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get_or_create(
+        config_entry_id=gateway.config_entry.entry_id, identifiers={("mac", "123")}
+    )
+
+    assert (
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6259301)